### PR TITLE
handle dynamic storage storageclass creation

### PR DESCRIPTION
**Pull request by laverya**
#### What this PR does / why we need it:

This allows installing with two openebs storageclasses and no rook storageclass, while still installing the rook operator.

It will be used by the scaling from openebs to rook feature in the future.

(you could use the rook param with this change, but it would not be particularly useful until the ekco changes in the pipeline are completed)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: addons/ekco/template/base/install.sh [hunk 1]
```diff
@@ -464,3 +466,14 @@ function ekco_load_images() {
         find "$DIR/image-overrides" -type f | xargs -I {} bash -c "cat {} | ctr -a $(${K8S_DISTRO}_get_containerd_sock) -n=k8s.io images import -"
     fi
 }
+
+function ekco_dynamicstorage() {
+    if [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```
Comment (emosbaugh) [inline addons/ekco/template/base/install.sh:471]: what if ROOK_MINIMUM_NODE_COUNT is not set? will it be empty? in this case dont you need to check that there is a value?

```suggestion
    if [ -n "$ROOK_MINIMUM_NODE_COUNT" ] && [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```

File: addons/openebs/template/base/install.sh [hunk 2]
```diff
@@ -291,6 +291,9 @@ function openebs_should_be_default_storageclass() {
     elif openebs_has_default_storageclass ; then
         # if there is already a default storage class that is not "$storage_class_name"
         return 1
+    elif [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```
Comment (emosbaugh) [inline addons/openebs/template/base/install.sh:294]:
```suggestion
    elif [ -n "$ROOK_MINIMUM_NODE_COUNT" ] && [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```

File: addons/rook/template/base/install.sh [hunk 3]
```diff
@@ -86,6 +86,11 @@ function rook() {
     rook_operator_deploy
     rook_set_ceph_pool_replicas
     rook_ready_spinner # creating the cluster before the operator is ready fails
+
+    if [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```
Comment (emosbaugh) [inline addons/rook/template/base/install.sh:90]:
```suggestion
    if [ -n "$ROOK_MINIMUM_NODE_COUNT" ] && [ "$ROOK_MINIMUM_NODE_COUNT" -gt "1" ]; then
```

File: addons/rook/template/base/install.sh [hunk 4]
```diff
@@ -86,6 +86,11 @@ function rook() {
...
+        return 0 # do not create a ceph cluster if it should instead be managed by ekco
```
Comment (emosbaugh) [inline addons/rook/template/base/install.sh:91]: below this is looks like there may be code that ekco will not manage. for example `rook_maybe_migrate_from_longhorn`

it looks like you took into account upgrades but im not entirely convinced upgrades will work in some scenarios.

Comment (laverya) [reply addons/rook/template/base/install.sh:91]: agreed this really doesn't handle migrating from longhorn right now

**Review by emosbaugh — DISMISSED** [review]

**Review by emosbaugh — APPROVED** [review]

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**